### PR TITLE
IBX-5084 Catalog Twig functions

### PR DESCRIPTION
**Pull request by dabrt**
| Question      | Answer |
| ------------- | --- |
| JIRA Ticket   | IBX-5084 |
| Versions      | 4.4 and up |
| Edition       | All |

Add a list of Twig functions related to product catalogs.

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [ ] Code samples are working
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/templating/twig_function_reference/catalog_twig_functions.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+# Catalog Twig functions
+
+With the catalog Twig functions you can get catalog location, render catalog status and list filter templates.
+
+### `ibexa_get_filter_preview_templates`
+
+The `ibexa_get_filter_preview_templates()` function gets a list of filter preview Twig templates 
+(configured in `ibexa.repositories.product_catalog.filter_preview_templates`). 
+Thjen it sorts the list by priority and renders the filter templates.
```
Comment (konradoboza) [inline docs/templating/twig_function_reference/catalog_twig_functions.md:13]:
```suggestion
Then it sorts the list by priority and renders the filter templates.
```

File: docs/templating/twig_function_reference/catalog_twig_functions.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                {{ block(block_name, template) }}
+
+                {% set block_found = true %}
+            {% endif %}
+        {% endfor %}
+```
+
+### `ibexa_get_product_catalog_root`
+
+The `ibexa_get_product_catalog_root()` function gets an ID of a location of the product catalog (configured in `ibexa_product_catalog.engines.default.type.options.root_location_remote_id`).
```
Comment (konradoboza) [inline docs/templating/twig_function_reference/catalog_twig_functions.md:27]:
```suggestion
The `ibexa_get_product_catalog_root()` function gets an ID of a root location of the product catalog (configured in `ibexa_product_catalog.engines.default.type.options.root_location_remote_id`).
```

**Review by konradoboza — CHANGES_REQUESTED** [review]

File: docs/templating/twig_function_reference/catalog_twig_functions.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+``` html+twig
+        {% for template in ibexa_get_filter_preview_templates() %}
+            {% if block(block_name, template) is defined %}
+                {{ block(block_name, template) }}
+
+                {% set block_found = true %}
+            {% endif %}
+        {% endfor %}
+```
```
Comment (adamwojs) [inline docs/templating/twig_function_reference/catalog_twig_functions.md:23]: @mikadamczyk Please correct me if I'm wrong but this function is consider as internal and doesn't have real use case in customer projects

File: docs/templating/twig_function_reference/catalog_twig_functions.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+### `ibexa_get_product_catalog_root`
+
+The `ibexa_get_product_catalog_root()` function gets a remote ID of a location of the product catalog (configured in `ibexa_product_catalog.engines.default.type.options.root_location_remote_id`).
```
Comment (adamwojs) [inline docs/templating/twig_function_reference/catalog_twig_functions.md:27]: `ibexa_get_product_catalog_root` returns product catalog root location, not "remote ID of a location of the product catalog"

File: docs/templating/twig_function_reference/catalog_twig_functions.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+
+``` html+twig
+{% set product_catalog_root_id = ibexa_get_product_catalog_root() %}
+
+{% set product_catalog_root = location(product_catalog_root_id) %}
+
+    /// Perform operations on the product catalog 
+```
```
Comment (adamwojs) [inline docs/templating/twig_function_reference/catalog_twig_functions.md:36]: There is no such Twig function like `location`

```
Unknown "location" function. Did you mean "ez_render_location", "ez_http_tag_location", "ez_path_to_locations", "ibexa_render_location", "ibexa_path_to_locations", "ez_http_cache_tag_location", "ibexa_http_cache_tag_location", "ez_http_tag_relation_location_ids", "ibexa_http_cache_tag_relation_location_ids"?
```

**Review by adamwojs — CHANGES_REQUESTED** [review]

File: docs/templating/twig_function_reference/catalog_twig_functions.md (hunk 4 above)
Comment (konradoboza) [reply docs/templating/twig_function_reference/catalog_twig_functions.md:27]: That's correct, I might have suggested the wrong description to @dabrt.

File: docs/templating/twig_function_reference/catalog_twig_functions.md (hunk 3 above)
Comment (mikadamczyk) [reply docs/templating/twig_function_reference/catalog_twig_functions.md:23]: I agree. For me, the only use case is when someone may want to create his own UI for the admin panel

File: docs/templating/twig_function_reference/catalog_twig_functions.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+### `ibexa_render_catalog_status`
+
+The `ibexa_render_catalog_status` filter renders the status of the catalog, translated into the current language.
+
+#### Examples
+
+``` html+twig
+{% set status_code = content.contentInfo.status %}
+{% set status_label = status_code|ibexa_render_catalog_status %}
```
Comment (mikadamczyk) [inline docs/templating/twig_function_reference/catalog_twig_functions.md:46]:
```suggestion
{% set status_label = catalog.status|ibexa_render_catalog_status %}
```

Comment (mikadamczyk) [reply docs/templating/twig_function_reference/catalog_twig_functions.md:46]: or with the usage of macro
```
{% import "@ibexadesign/product_catalog/catalog_macros.html.twig" as catalog_macros %}
{{ catalog_macros.status_node(catalog.status) }}

**Review by konradoboza — APPROVED** [review]

File: docs/templating/twig_function_reference/twig_function_reference.md [hunk 7]
```diff
@@ -10,13 +10,14 @@ and [Twig extensions provided by Symfony]([[= symfony_doc =]]/reference/twig_ref
 
 [[= cards([
   "templating/twig_function_reference/cart_twig_functions",
+  "templating/twig_function_reference/catalog_twig_functions",
```
Comment (juskora) [inline docs/templating/twig_function_reference/twig_function_reference.md:13]: Broken link.

**Review by juskora — CHANGES_REQUESTED** [review]

**Review by juskora — APPROVED** [review]

**Pull request merged**